ui/src/ducks/login.js

```diff
@@ -28,23 +36,19 @@ export default function reducer(state = defaultState, action = {}) {
 }
 
 // Action Creators
-export const authenticateSaltApiAction = payload => {
-  return { type: AUTHENTICATE_SALT_API, payload };
-};
-
-export const setSaltAuthenticationSuccessAction = payload => {
+export const setSaltAuthenticationSuccessAction = (payload: ApiSalt.SaltToken) => {
   return {
     type: SALT_AUTHENTICATION_SUCCESS,
     payload,
   };
 };
 
 // Sagas
-export function* authenticateSaltApi() {
-  const api = yield select(apiConfigSelector);
-  const user = yield select(state => state.oidc.user);
-  const result = yield call(ApiSalt.authenticate, user);
-  if (!result.error) {
+export function* authenticateSaltApi(): Generator<any, void, any> {
+  const api: Config | null = yield select(apiConfigSelector);
+  const user: User = yield select((state: RootState) => state.oidc?.user);
+  const result: { error: any } | ApiSalt.SaltToken = yield call(ApiSalt.authenticate, user);
```

Comment:
Could be handy to have type aliases for such "result" values:

```javascript
declare type Result<T> = { error: any } | T

// Or for API results under a `body` key
declare type APIResult<T> = Result<{ body: T }>
```

Here we'd have:

```javascript
const result: Result<Api.SaltToken> = yield call(...)
```